channeldb/payments.go

```diff
@@ -601,9 +604,28 @@ func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
 				totalPayments uint64
 				err           error
 			)
-			countFn := func(_, _ []byte) error {
-				totalPayments++
 
+			countFn := func(sequenceKey, hash []byte) error {
```

Comment:
This func won't work in SQL unless we change `ForAll` to `ForEach` below.

Reply:
I did the change ... now it is working even to SQL databases ... but as you said there is no otimization when there is a SQL database available because we cannot use the `ForAll` instead of `ForEach` as we done before this fix.

As you said below maybe we shouldn't touch it until we can find an efficient counting method. Maybe this is another PR to come!